app/ts/App.tsx

```diff
@@ -92,6 +125,34 @@ export function App() {
 	const maybeAccountAddress = useOptionalSignal<AccountAddress>(undefined)
 	const chainId = useSignal<number | undefined>(undefined)
 	const inputTimeoutRef = useRef<number | null>(null)
+	const universe = useOptionalSignal<AccountAddress>(undefined)
+	const universeForkingInformation = useOptionalSignal<Awaited<ReturnType<typeof getUniverseForkingInformation>>>(undefined)
+	const activeTab = useSignal(0)
+
+	const tabs = [
+		{ title: 'Trading', path: 'trading', component: <DeployContract maybeAccountAddress = { maybeAccountAddress } areContractsDeployed = { areContractsDeployed }/> },
+		{ title: 'Market Creation', path: 'market-creation', component: <CreateYesNoMarket maybeAccountAddress = { maybeAccountAddress }/> },
+		{ title: 'Reporting', path: 'reporting', component: <Reporting maybeAccountAddress = { maybeAccountAddress }/> },
+		{ title: 'Claim Funds', path: 'claim-funds', component: <ClaimFunds maybeAccountAddress = { maybeAccountAddress }/> }
+	] as const
+
+	useEffect(() => {
+		const hash = window.location.hash.replace('#/', '')
+		const [path, params] = hash.split('?')
+		const tabIndex = tabs.findIndex(tab => tab.path === path)
+		if (tabIndex !== -1) {
+			activeTab.value = tabIndex
+		}
```

Comment:
Why did you split the tab handling between the Tabs component and the App?  It feels like all of this should live in the Tabs component like you had it before.

Reply:
I wanted the information on which universe we are in outside the tab. We could do the url processing (the router) in two places, but I feel its clearer if its in one place

Reply:
Can you make your `Tabs` component accept an `OptionalSignal<'trading'|'market-creation'|'reporting'|'claim-funds'>` for current route, that way you can have access to the route outside of the component but all management of it is handled by the component?

I recommend making the signal a `ReadOnlyOptionalSignal` outside of the component and only writable by the component, to ensure that you don't accidentally mess with it elsewhere.

Alternatively, you could make it so anyone can update the signal and when that signal changes it will trigger a route update.

Reply:
not sure how we could do that as then changing the tab should also change the routing. If you click "trading" the url should change to "trading" too

Reply:
I added issue for this, I think I need to add stuff to router anyway later